lib/pure/algorithm.nim

```diff
@@ -228,7 +228,8 @@ proc smartBinarySearch*[T](a: openArray[T], key: T): int {.deprecated:
 const
   onlySafeCode = true
 
-proc lowerBound*[T, K](a: openArray[T], key: K, cmp: proc(x: T, k: K): int {.closure.}): int =
+proc lowerBound*[T, K](a: openArray[T], key: K, cmp: proc(x: T, k: K): int {.
```

Comment:
I'll live with this but we should debate it, I would strongly prefer:

```nim
proc lowerBound*[T, K](
  a: openArray[T],
  key: K,
  cmp: proc(x: T, k: K): int {.closure.}
): int =
```

or something along those lines instead of this strange split we have right now.